miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java

```diff
@@ -99,6 +118,134 @@ protected BoxDto asDto(Box model) {
     return Dtos.asBoxDtos(results, true);
   }
 
+  /**
+   * Creates an Excel spreadsheet that contains the list of all Boxable items located in a particular position. Empty positions are not
+   * listed.
+   *
+   * @param boxId ID of the Box
+   * @return JSON object with "hashCode" field representing the hash code of the spreadsheet filename
+   */
+  @RequestMapping(value = "/box/{boxId}/spreadsheet", method = RequestMethod.GET)
+  public @ResponseBody JSONObject createSpreadsheet(@PathVariable("boxId") Long boxId) {
+    try {
+      return exportBoxContentsForm(boxId);
+    } catch (Exception e) {
+      throw new RestException("Failed to get contents form", Status.BAD_REQUEST);
+    }
+  }
+
+  private JSONObject exportBoxContentsForm(Long boxId) throws IOException  {
+    Box box = boxService.get(boxId);
+
+    List<List<String>> boxContents = getBoxContents(box);
+    String name = box.getName();
+    String alias = box.getAlias();
+
+    File f = misoFileManager.getNewFile(Box.class, "forms", "BoxContentsForm-" + getCurrentDateAsString() + ".xlsx");
+    if (detailedSample) {
+      FormUtils.createDetailedBoxSpreadsheet(f, name, alias, boxContents);
+    } else {
+      FormUtils.createPlainBoxSpreadsheet(f, name, alias, boxContents);
+    }
+
+    return JSONObject.fromObject("{hashCode: " + f.getName().hashCode() + "}");
+  }
+
+  /**
+   * @param box
+   * @return List of lists of strings, where each list of string represents a box
+   */
+  private List<List<String>> getBoxContents(Box box) throws IOException {
+    List<List<String>> boxContents = new ArrayList<>();
+
+    // Use TreeMap to iterate through the contents of the box in order of box position
+    for (Map.Entry<String, BoxableView> entry : new TreeMap<>(box.getBoxables()).entrySet()) {
+      String position = entry.getKey();
+      BoxableView boxableView = entry.getValue();
+
+      String name = boxableView.getName();
+      String alias = boxableView.getAlias();
+      String barcode = boxableView.getIdentificationBarcode();
+
+      if (detailedSample) {
+        String externalName = findExternalName(boxableView);
+        String numSlides = findNumSlides(boxableView);
+
+        boxContents.add(Arrays.asList(position, name, alias, barcode, externalName, numSlides));
+      } else {
+        boxContents.add(Arrays.asList(position, name, alias, barcode));
+      }
+    }
+
+    return boxContents;
+  }
+
+  private String findExternalName(BoxableView boxableView) throws IOException {
+    DetailedSample detailedSample;
+
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "detailedSample" which hides the field declared at line 53. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)